pkg/cmd/extension/http.go

```diff
@@ -13,32 +13,49 @@ import (
 	"github.com/cli/cli/v2/internal/ghrepo"
 )
 
-func hasScript(httpClient *http.Client, repo ghrepo.Interface) (hs bool, err error) {
+func repoExists(httpClient *http.Client, repo ghrepo.Interface) (bool, error) {
+	url := fmt.Sprintf("%s%s/%s", ghinstance.HostPrefix(repo.RepoHost()), repo.RepoOwner(), repo.RepoName())
```

Comment:
`ghinstance.HostPrefix` seems to be the correct one to use instead of `ghinstance.RESTPrefix` as I don't think we can rely on `api.` for private installs. However this returns a 404 for private repos (I tested with one I know works inside the GitHub org). I thought this was perhaps due to the auth header not being set but it seems like it is. Would appreciate someone's help fixing this!

Reply:
@mntlty I think we are already relying on `ghinstance.RESTPrefix` in many places and as far as I know have not heard any feedback about it not working for private installs. Is there something special about this endpoint that makes you think it won't work?

Reply:
Rest prefix does work for private installs! My concern was around a private installation, along the lines of example.com in the tests. I don't know exactly what is supported by the cli in terms of alternate hosts, if it looks good to you I'll switch it to that 🙏